storage/innobase/handler/ha_innodb.cc

```diff
@@ -21528,11 +21528,20 @@ update_innodb_redo_log_encrypt(
 		return;
 	}
 
-	byte iv[ENCRYPTION_KEY_LEN];
-	Encryption::random_value(iv);
+	ut_ad(strlen(server_uuid) > 0);
+
+	if (!Encryption::check_keyring()) {
+		ib_senderrf(
+			thd, IB_LOG_LEVEL_WARN,
```

Comment:
same as other PR comment, see if we can make it ERROR instead of WARN

Reply:
As we discussed on slack: I can make most of the messages errors, but not all of them - we can't report errors from the update function, only from validate. I could move most of the checks to validate, but it'll be more complex there, and I still would have to keep some warnings. So it's best to leave these as is.

Reply:
It doesn't make sense to move the code to validate() because of the limitation.

This is upstream limitation that we cannot return errors from SET's update() function. Can you please create bug report as Feature Request?